libnetwork/sandbox_dns_unix.go

```diff
@@ -181,6 +181,18 @@ func (sb *sandbox) setExternalResolvers(content []byte, addrType int, checkLoopb
 	}
 }
 
+// isIPv4Loopback checks if the given IP address is an IPv4 loopback address.
+// It's based on the logic in Go's net.IP.IsLoopback(), but only the IPv4 part:
+// https://github.com/golang/go/blob/go1.16.6/src/net/ip.go#L120-L126
+func isIPv4Loopback(ipAddress string) bool {
```

Comment:
This is a lot less eyebrow raising than the regular expressions above for parsing IP addresses in a language with a pretty decent standard library for parsing IP addresses. 😅 👍

Reply:
Ha! True. Yes, I didn't compare performance between this and the original approach, but I assume the stdlib code is usually optimised.